packages/core/src/services/scales-cartesian.ts

```diff
@@ -226,10 +230,14 @@ export class CartesianScales extends Service {
 
 	protected findMainHorizontalAxisPosition() {
 		const options = this.model.getOptions();
-		const axisOptions = Tools.getProperty(options, "axes");
+		const axesOptions = Tools.getProperty(options, "axes");
 
 		// If top axis has been specified as `main`
-		if (Tools.getProperty(axisOptions, AxisPositions.TOP, "main") === true) {
+		if (
+			(Tools.getProperty(axesOptions, AxisPositions.BOTTOM) === null &&
+				Tools.getProperty(axesOptions, AxisPositions.TOP) !== null) ||
+				Tools.getProperty(axesOptions, AxisPositions.TOP, "main") === true
+		) {
```

Comment:
For maximum verbosity, you could go with
```suggestion
		if ((
				Tools.getProperty(axesOptions, AxisPositions.BOTTOM) === null
				&& Tools.getProperty(axesOptions, AxisPositions.TOP) !== null
			)
			|| Tools.getProperty(axesOptions, AxisPositions.TOP, "main") === true
		) {
```
or something along those lines